### PR TITLE
PP-5632: rate-limiting follow-up changes

### PR DESCRIPTION
**Pull request by jankowiakmaria**
Addressing review comments (https://github.com/alphagov/pay-publicapi/pull/701 and https://github.com/alphagov/pay-publicapi/pull/697):
* filtering empty values in StringToListConverter
* using method from rateLimiterKey in RateLimiterFilter
* removing unused imports from RateLimiterKey
* using HttpMethod.POST instead of string in RateLimitManager
* using removeEnd to remove trailing slash in PathHelper
* refactor LocalRateLimiter to use RateLimiterKey

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/app/config/RateLimiterConfig.java [hunk 1]
```diff
@@ -27,6 +28,7 @@
     @Min(1)
     private int noOfReqForPostPerNode;
 
+    @Valid
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/app/config/RateLimiterConfig.java]: We need more than this. See https://github.com/alphagov/pay-connector/blob/86317f2faea6c1024d61fc4f7e3005f63d50e2fe/src/main/java/uk/gov/pay/connector/app/SqsConfig.java#L9 for example

File: src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java [hunk 2]
```diff
@@ -44,7 +44,8 @@ synchronized void checkRateOf(String accountId, RateLimiterKey key)
         if (count != null) {
             int allowedNumberOfRequests = rateLimitManager.getAllowedNumberOfRequests(key, accountId);
             if (count > allowedNumberOfRequests) {
-                LOGGER.info(String.format("RedisRateLimiter - Rate limit exceeded for account [%s] and type [%s] - count: %d, rate allowed: %d", accountId, key.getKeyType(), count, allowedNumberOfRequests));
+                LOGGER.info(String.format("RedisRateLimiter - Rate limit exceeded for account [%s] and method [%s] - count: %d, rate allowed: %d",
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java]: Unused import `javax.ws.rs.HttpMethod;`

Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java]: removed

File: src/main/java/uk/gov/pay/api/app/config/RateLimiterConfig.java (hunk 1 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/app/config/RateLimiterConfig.java]: Not needed. Each `noOf...` has `@Min(1)` validation which will cause the app to fail during the startup if the value is not provided (or 0).

**Pull request merged**